### PR TITLE
90 2j add localisation

### PR DESCRIPTION
**Pull request by Neyross**

### CODE REVIEW

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 1]
```diff
@@ -1,17 +1,25 @@
 package com.bouboulecorp.thurii.HomePage.HomePageFragments
 
+import android.Manifest
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:3]: 🚫 **[ktlint]** <import-ordering> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Imports must be ordered in lexicographic order without any empty lines in-between with "java", "javax", "kotlin" and aliases in the end

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
 import android.os.Bundle
 import androidx.preference.PreferenceManager
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.app.ActivityCompat
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:10]: 🚫 **[ktlint]** <no-unused-imports> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
+import androidx.core.content.ContextCompat
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:11]: 🚫 **[ktlint]** <no-unused-imports> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
+import androidx.core.content.PermissionChecker.PERMISSION_GRANTED
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:12]: 🚫 **[ktlint]** <no-unused-imports> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 5]
```diff
@@ -21,25 +29,43 @@ class MapFragment : Fragment(R.layout.fragment_map) {
...
 
-        val rotationGestureOverlay = RotationGestureOverlay(map)
-        rotationGestureOverlay.isEnabled
-        map.setMultiTouchControls(true)
-        map.overlays.add(rotationGestureOverlay)
+        val locationOverlay = MyLocationNewOverlay(GpsMyLocationProvider(context), map)
+        locationOverlay.enableMyLocation()
+        locationOverlay.enableFollowLocation()
+        locationOverlay.isDrawAccuracyEnabled = true
+        locationOverlay.runOnFirstFix(){ (context as MainActivity).runOnUiThread {
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:61]: 🚫 **[ktlint]** <curly-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing spacing before "{"

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:61]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline after "{"

File: app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt [hunk 6]
```diff
@@ -21,25 +29,43 @@ class MapFragment : Fragment(R.layout.fragment_map) {
...
+            mapController.setCenter(locationOverlay.myLocation)
+            mapController.setZoom(16.0) // has to be a double
+        }}
```
Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:64]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline before "}"

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:64]: 🚫 **[ktlint]** <curly-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing spacing after "}"

Comment (Neyross) [inline app/src/main/java/com/bouboulecorp/thurii/HomePage/HomePageFragments/MapFragment.kt:64]: 🚫 **[ktlint]** <curly-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing spacing before "}"

**Review by larroyo3 — CHANGES_REQUESTED** [review]
plusieurs problème du linter

**Review by larroyo3 — APPROVED** [review]
:ok_hand:

**Review by claraxuxu — APPROVED** [review]

**Pull request merged**